### ISSUE TITLE
@parcel/validator-typescript fails to typecheck.

### ISSUE THREAD

**Issue by astegmaier** [parcel-bundler/parcel#4138]
# 🐛 bug report

After configuring `@parcel/validator-typescript` with a `.parcelrc` file, it fails to report on type errors

(This is related to #4136 - it's the issue that appeared after I fixed the problem created there).

## 🎛 Configuration (.babelrc, package.json, cli command)

***.parcelrc:***
```js
{
    "extends": "@parcel/config-default",
    "validators": {
        "*.{ts,tsx}": ["@parcel/validator-typescript"]
    }
}
```
***package.json:***
```
{
  "name": "parcel2-typescript-validator-bug",
  "version": "1.0.0",
  "description": "A demonstration of a bug when attempting to use @parcel/validator-typescript",
  "main": "index.js",
  "repository": "https://github.com/astegmaier/parcel2-typescript-validator-bug.git",
  "author": "Andrew Stegmaier <ansteg@microsoft.com>",
  "license": "MIT",
  "private": true,
  "scripts": {
    "build": "parcel build src/index.ts",
    "start": "parcel src/index.ts"
  },
  "devDependencies": {
    "@parcel/validator-typescript": "2.0.0-nightly.112",
    "parcel": "2.0.0-nightly.110",
    "typescript": "^3.7.5"
  }
}
```
***src/index.ts:***
```ts
const message: number = "Hello world!"; // This is a type error!
console.log(message);
```

## 🤔 Expected Behavior

Running `yarn build` or `yarn start` should eventually show an error.

## 😯 Current Behavior

No type errors are displayed.

## 💁 Possible Solution

I'm stumped. 

Interesting data point: when I had the configuration from #4136 - using version `2.0.0-alpha.3` of `@parcel/validators` and then modified the installed `package.json` for the `@parcel/ts-utils` package (see that issue), I _do_ get type errors. This suggests whatever is the problem was probably a relatively recent change.

## 🔦 Context

Trying to get parcel2 working end-to-end in a typescript-based project, and I'm super excited about using the built-in typechecker (with awesome error reporting 😃 )

## 💻 Code Sample

See [this sample project demonstrating the repo](https://github.com/astegmaier/parcel2-typescript-validator-bug/tree/new-issue).

## 🌍 Your Environment

| Software         | Version(s) |
| ---------------- | ---------- |
| Parcel           | 2.0.0-nightly.105
| Node             | 10.6.3
| Yarn         | 1.21.1
| Operating System | OSX 10.15.3

**Comment by DeMoorJasper** [comment]
The typescript validator requires you to have a `tsconfig.json` before it takes effect.

https://github.com/parcel-bundler/parcel/blob/v2/packages/validators/typescript/src/TypeScriptValidator.js#L41-L42

Maybe we should default the config to an empty object...

**Comment by astegmaier** [comment]
Thanks for the tip! That unblocked me for now. I agree that for the best getting started experience, it should probably have a default configuration that doesn't require a `tsconfig.json`. I'm happy to submit a PR for this.

**Comment by DeMoorJasper** [comment]
@astegmaier yeah I agree, it might be confusing to people that it doesn't work without a tsconfig, definitely feel free to try fix it and open a PR

**Comment by matthewoates** [comment]
The docs are very misleading with TypeScript support and at the moment I regret switching to parcel 2.

As a stop-gap before I switch to something else or parcel 2 properly supports typescript, I'm using `tsc --noEmit --watch` to surface errors. Of course this is not ideal because I have to keep this process running and watch the window for errors.'

Does anyone have a better solution at the moment?
